state/konnect_types.go

```diff
@@ -12,9 +12,41 @@ type Document struct {
 	Meta
 }
 
+// Identifier returns the endpoint key name or ID.
+func (d1 *Document) Identifier() string {
+	if d1.Path != nil {
+		return *d1.Path
+	}
+	return *d1.ID
+}
+
 // Console returns an entity's identity in a human-readable string.
-func (d *Document) Console() string {
-	return *d.Path
+func (d1 *Document) Console() string {
+	return *d1.Path
+}
+
+// Equal returns true if s1 and s2 are equal.
+func (d1 *Document) Equal(d2 *Document) bool {
+	return d1.EqualWithOpts(d2, false, false, false)
+}
+
+// EqualWithOpts returns true if d1 and d2 are equal.
+// If ignoreID is set to true, IDs will be ignored while comparison.
+// If ignoreTS is set to true, timestamp fields will be ignored.
+func (d1 *Document) EqualWithOpts(d2 *Document,
```

Comment:
The problem of comparing structs with certain fields ignored has been solved already: [go-cmp](https://pkg.go.dev/github.com/google/go-cmp/cmp) with `FilterPath` and `Ignore` can do that and I've successfully used that in other projects.

Not requesting changes here - sharing for visibility of the alternative, potentially better, approach.